File: dart/2155.dart
class Solution {
    List<int> maxScoreIndices(List<int> n) {
        int r = 0, l = 0, q = n.length, m = -1;
        List<int> v = [];
        for (int i in n)
            r += i;
        for (int i = 0; i < q; i++) {
            if (r + l > m) {
                m = l + r;
                v.clear();
            }
            if (r + l >= m)
                v.add(i);
            l += 1 - n[i];
            r -= n[i];
        }
        if (r + l > m) {
            m = l + r;
            v.clear();
        }
        if (r + l >= m)
            v.add(q);
        return v;
    }
}

File: python3/2155.py
class Solution:
    def maxScoreIndices(self, n: List[int]) -> List[int]:
        l = 0
        r = Counter(n).get(1, 0)
        m = -1
        q = len(n)
        v = []
        for i in range(q):
            if r + l > m:
                m = l + r
                v = []
            if r + l >= m:
                v.append(i)
            l += 1 - n[i]
            r -= n[i]
        if r + l > m:
            return [q]
        if r + l == m:
            return v + [q]
        return v
